subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalInputsIntegrationTest.groovy

```diff
@@ -73,4 +105,45 @@ class IncrementalInputsIntegrationTest extends AbstractIncrementalTasksIntegrati
         executesWithRebuildContext()
     }
 
+    def "cannot query non-incremental file input parameters"() {
+        given:
+        previousExecution()
+
+        when:
+        file("inputs/new-input-file.txt") << "new file"
+        buildFile << """
+            tasks.withType(IncrementalTask).configureEach {
+                queryChangesFor.add("nonIncrementalInput")
+            }
+        """
+        then:
+        fails("incremental")
+        failure.assertHasCause("Cannot query incremental changes: No property found for value ${file("nonIncrementalInput").absolutePath}. Incremental properties: anotherIncrementalInput, inputDir.")
+    }
+
+    def "changes to non-incremental input parameters cause a rebuild"() {
+        given:
+        file("nonIncrementalInput").makeOlder()
+        previousExecution()
+
+        when:
+        file("inputs/new-input-file.txt") << "new file"
+        file("nonIncrementalInput").text = 'changed'
```

Comment:
It would be good to split this into two tests to see each change triggering non-incremental execution.

Reply:
I don't understand. Only the second change causes a rebuild, the incremental change doesn't. Note that we have some more tests in the base class, which check changes on the incremental input already.